backend/lib/endpoints/posts.js

```diff
@@ -21,135 +22,355 @@ async function routes(app) {
   const { mongo } = app;
   const Comment = mongo.model("Comment");
   const Post = mongo.model("Post");
+  const User = mongo.model("User");
 
   // /posts
+  const UNLOGGED_POST_SIZE = 120;
 
-  app.get("/", { schema: getPostsSchema }, async (req) => {
-    // todo: add limit, skip, visibility filter, needs, tags ...
-    const { authorId } = req.params;
-    const aggregates = authorId ? [{ $match: { authorId } }] : [];
-    aggregates.push(
-      {
-        $lookup: {
-          as: "comments",
-          foreignField: "postId",
-          from: "comments",
-          localField: "_id",
+  app.get(
+    "/",
+    {
+      preValidation: [app.authenticate],
+      schema: getPostsSchema,
+    },
+    async (req) => {
+      const { userId } = req.query;
+      const [userErr, user] = await app.to(User.findById(userId));
+      if (userErr) {
+        throw app.httpErrors.notFound();
+      }
+
+      // Base filters - expiration and visibility
+      /* eslint-disable sort-keys */
+      const filters = [
+        { $or: [{ expireAt: null }, { expireAt: { $gt: new Date() } }] },
+        {
+          $or: [
+            { visibility: "worldwide" },
+            {
+              visibility: "country",
+              "author.location.country": user.location.country,
+            },
+            {
+              visibility: "state",
+              "author.location.country": user.location.country,
+              "author.location.state": user.location.state,
+            },
+            {
+              visibility: "city",
+              "author.location.country": user.location.country,
+              "author.location.state": user.location.state,
+              "author.location.city": user.location.city,
+            },
+          ],
         },
-      },
-      {
-        $project: {
-          _id: true,
-          commentsCount: {
-            $size: "$comments",
+      ];
+      /* eslint-enable sort-keys */
+
+      // Additional filters
+      const { paramFilters } = req.params;
+      if (paramFilters) {
+        // TODO: additional filters
+      }
+
+      // Unlogged user limitation for post content size
+      // TODO: how to check for logged/unlogged user?
+      /* eslint-disable sort-keys */
+      const contentProjection = user
```

Comment:
I recall a discussion we had in the engineering meeting last weekend where we were going to truncate the post on the frontend rather than the backend, are we not doing that anymore?

Reply:
Also, if we have the authenticate hook on this endpoint, then I don't think unauthenticated users would be able to get this this point since the authenticate hook will throw an error.

Reply:
I was also not sure if that was to be done in the backend or frontend. Talking with @anniegiang I decided to leave it done if we decided to go with backend strategy since this endpoint has different behaviors based on whether the user is authenticated or not. 

Anyway, if we decided to go the fronted route we can still just remove this project from the pipeline and the auth hook here.

Reply:
@manualzuru suggested doing this on frontend, I think for the feed for MVP this is fine, also yes I think we already discussed this a few times and it was clear that the auth is not needed for reading posts.